src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -690,6 +691,18 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 DiagnosticLevel.Error,
                 "BCP119",
                 $"Unsupported scope for extension resource deployment. Expected a resource reference.");
+
+            public Diagnostic ResourceMultipleDeclarations(string resourceNames) => new(
+                TextSpan,
+                DiagnosticLevel.Warning,

```

Comment:
Warning will allow compilation to continue. Should this be `DiagnosticLevel.Error`?

Reply:
This is an open question - I think warning as properly built ARM JSON will be built (no compilation blocker) and problem will be on deployment (runtime)

Reply:
We should always block compilation if we know we're going to generate something that is invalid. Warnings are just for scenarios where it's not possible to say whether or not the template is valid, or if we're trying to advise against a bad practice. So in this scenario, since we know for sure the template won't deploy, we should block it.

Reply:
Ok, will change to errors.

Although question on showing error itself - should I highlight only value (`'resource-name'`) that is invalid or entire property (`name: 'resource-name'`)?

Reply:
I think only the value, as there is nothing wrong with the property called `name`, just the value of it. Same as we do for other property values:

![image](https://user-images.githubusercontent.com/11078275/103418943-84aaf880-4b56-11eb-902b-b82c3e90a192.png)

Reply:
done.